src/models/loan/paymentCalculations.js

```diff
@@ -0,0 +1,46 @@
+function daysInYear(year) {
+  return 365
+}
+
+function daysInMonth(month) {
+  return 30.417
+}
+
+function effectiveMonthlyInterestRate(yearlyInterestRate, month, year) {
+  return ((1 + yearlyInterestRate / daysInYear(year)) ** daysInMonth(month)) - 1
+}
+
+const calculateRepaymentTerm = ({debt:{amountOwed, interestRate: {rate}}, paymentPlan: {monthlyPayment}}) => {
+  let repaymentTerm
+  const interestRate = effectiveMonthlyInterestRate(rate)
+  if (amountOwed == 0 ) {
+    repaymentTerm = 0
+  } else if (rate) {
+    if (monthlyPayment <= amountOwed * interestRate && amountOwed > 0) {
+      repaymentTerm = Infinity
+    } else {
+      repaymentTerm = (Math.log(monthlyPayment)
+                        - Math.log(monthlyPayment - amountOwed * effectiveMonthlyInterestRate(rate))
+                      ) / Math.log(1 + effectiveMonthlyInterestRate(rate))
+    }
+  } else {
+    repaymentTerm = amountOwed / monthlyPayment
+  }
+
+  return Math.ceil(repaymentTerm)
+}
+
+const calculateMonthlyPayment = ({debt:{amountOwed, interestRate: {rate}}, paymentPlan: {repaymentTerm}}) => {
+  let monthlyPayment
+  if (amountOwed == 0) {
+    monthlyPayment = 0
+  } else if (rate) {
+    const interestRate = effectiveMonthlyInterestRate(rate)
+    monthlyPayment = amountOwed * (interestRate / (1 - (1 + interestRate) ** -repaymentTerm))
+  } else {
+    monthlyPayment = amountOwed / repaymentTerm
+  }
+  return monthlyPayment
```

Comment:
similar to my comment above, you could just return out each of the statements instead of mutating a variable. you could also remove the last `else` that way:
```javascript
if (..) {
  ...
  return
} else if (rate) {
  ...
  return
}
return 
```